driver-core/src/main/com/mongodb/client/model/changestream/ChangeStreamDocument.java

```diff
@@ -232,7 +266,30 @@ public String getDatabaseName() {
     }
 
     /**
-     * Returns the fullDocument
+     * Returns the fullDocument.
+     *
+     * <p>
+     * Always present for operations of type {@link OperationType#INSERT} and {@link OperationType#REPLACE}. Also present for operations
+     * of type {@link OperationType#UPDATE} if the user has specified {@link FullDocument#UPDATE_LOOKUP} for the {@code fullDocument}
+     * option when creating the change stream.
+     * </p>
+     *
+     * <p>
+     * For operations of type {@link OperationType#INSERT} and {@link OperationType#REPLACE}, the value will contain the document being
+     * inserted or the new version of the document that is replacing the existing document, respectively.
+     * </p>
+     *
+     * <p>
+     * For operations of type {@link OperationType#UPDATE}, the value will contain a copy of the full version of the document from some
+     * point after the update occurred. If the document was deleted since the updated happened, the value will be null.
```

Comment:
Suggestion: the value will be null -> the value _may_ be null.

Reply:
The spec has it as `will`, but I think `may` is closer to the truth.  Changed.

Reply:
Cool - its all about when the delete event occurs.